### ISSUE TITLE
[Bug] [Tools] Connecting to upstream json-rpc by default causes unexplained app errors

### ISSUE THREAD

**Issue by lpgeiger** [OLSF/libra#576]
TL;DR the apps in ./ol/. give the false impression of connecting to the localhost (the local node), when usually they are connecting to an upstream. This is because of boot order issues, using `ol start`

What's the problem:
Many nodes are connecting to an upstream instead of localhost for jsonrpc. This creates a problem where if many nodes are using the same upstream e.g. 167.x.x.x , whenever it goes into maintenance mode the apps of downstream nodes may fail unexpectedly.

Why this happens:
The `ol` tool checks a localhost and an upstream to create a LibraClient. The issue is usually on `ol start` or `ol serve` the client that will be found is the upstream, and not usually the localhost (since it may not have yet booted). So the command will keep using an upstream instead of localhost. This is especially the case with `ol serve` which will use an upstream to bootup, but never retry with a localhost. 

Suggested fix:
So the default behavior for ol would be: try the localhost first, then fallback to upstream. And if it is using the upstream, it should periodically try to reconnect to the localhost.

related: https://github.com/OLSF/libra/issues/621

**Comment by lpgeiger** [comment]
Merged with related issue https://github.com/OLSF/libra/projects/6?card_filter_query=milestone%3A0lv5
